server/mediaserver.go

```diff
@@ -673,19 +677,60 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Do the transcoding!
-	err = processSegment(cxn, seg)
+	tr, err := processSegment(cxn, seg)
 	if err != nil {
 		// TODO return error
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	if tr == nil {
+		http.Error(w, "No sessions available", http.StatusServiceUnavailable)
+		return
+	}
 
+	boundary := common.RandName()
+	accept := r.Header.Get("Accept")
+	if accept == "multipart/mixed" {
+		contentType := "multipart/mixed; boundary=" + boundary
+		w.Header().Set("Content-Type", contentType)
+	}
 	w.WriteHeader(http.StatusOK)
+	w.(http.Flusher).Flush()
+	if accept == "multipart/mixed" {
```

Comment:
The level of indentation here is getting kind of deep, we could probably save one level of indentation if we return from the function if not multipart, eg

```golang
if accept != "multipart/mixed" {
  return
}
mw := multipart.NewWriter(w)
...
```

(If we ever add another content type then we can factor things out, and the nice thing is that would also lead to a cleaner diff since it likely wouldn't involve re-indentation if the refactoring meant putting each output format into its own function. But the sketch above is OK for now.)

Reply:
fixed